randovania/generator/generator.py

```diff
@@ -333,3 +341,48 @@ def actual_status_update(msg: str) -> None:
             )
 
     return result
+
+
+def require_all_majors(configuration: BaseConfiguration, game: GameDescription):
+    resources = ResourceCollection.with_database(game.resource_database)
+
+    # Progression
+    for pickup, state in configuration.standard_pickup_configuration.pickups_state.items():
```

Comment:
You should not be creating entirely new pickups for this and instead use the same list of pickups that is sent to generator. Probably around line 150 of this file is the best place